{"name": "Номер заказа отображается в разделе В работе после оформления заказа", "status": "broken", "statusDetails": {"message": "Exception: Номер заказа не изменился!", "trace": "self = <pages.main_page.MainPage object at 0x107337810>\nlocator = ('xpath', \"//h2[@class='Modal_modal__title_shadow__3ikwq Modal_modal__title__2L34m text text_type_digits-large mb-8']\"), time = 20\n\n    def wait_for_order_number_change(self, locator, time=20):\n        def _order_number_changed(driver):\n            element = self.find_element_located_visibility(locator)\n            order_number = element.text\n            return order_number != '9999'\n    \n        try:\n>           WebDriverWait(self.browser, time).until(_order_number_changed)\n\npages/base_page.py:67: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = <selenium.webdriver.support.wait.WebDriverWait (session=\"053d0c61-b4a8-41f0-b05b-e788f98c30dd\")>\nmethod = <function BasePage.wait_for_order_number_change.<locals>._order_number_changed at 0x10731efc0>, message = ''\n\n    def until(self, method: Callable[[D], Union[Literal[False], T]], message: str = \"\") -> T:\n        \"\"\"Calls the method provided with the driver as an argument until the \\\n        return value does not evaluate to ``False``.\n    \n        :param method: callable(WebDriver)\n        :param message: optional message for :exc:`TimeoutException`\n        :returns: the result of the last call to `method`\n        :raises: :exc:`selenium.common.exceptions.TimeoutException` if timeout occurs\n        \"\"\"\n        screen = None\n        stacktrace = None\n    \n        end_time = time.monotonic() + self._timeout\n        while True:\n            try:\n                value = method(self._driver)\n                if value:\n                    return value\n            except self._ignored_exceptions as exc:\n                screen = getattr(exc, \"screen\", None)\n                stacktrace = getattr(exc, \"stacktrace\", None)\n            time.sleep(self._poll)\n            if time.monotonic() > end_time:\n                break\n>       raise TimeoutException(message, screen, stacktrace)\nE       selenium.common.exceptions.TimeoutException: Message:\n\n../../page_object_pythonProject/.venv/lib/python3.11/site-packages/selenium/webdriver/support/wait.py:105: TimeoutException\n\nDuring handling of the above exception, another exception occurred:\n\nself = <tests.test_order_feed_page.TestOrderFeedPage object at 0x107211710>\nbrowser = <selenium.webdriver.firefox.webdriver.WebDriver (session=\"053d0c61-b4a8-41f0-b05b-e788f98c30dd\")>\n\n    @allure.title(\"Номер заказа отображается в разделе В работе после оформления заказа\")\n    def test_check_order_number_in_the_work_panel_after_making_order(self, browser):\n        main_page = MainPage(browser)\n        main_page.go_to_site(URLS.MAIN_PAGE_URL)\n        main_page.click_login_button()\n        login_page = LoginPage(browser)\n        login_page.input_email_field()\n        login_page.input_password_field()\n        login_page.click_enter_button()\n        main_page.drag_r2_d3_bun_to_constructor()\n        main_page.click_make_order_button()\n>       main_page.wait_for_9999_number_disappear_on_created_order_window()\n\ntests/test_order_feed_page.py:85: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \npages/main_page.py:70: in wait_for_9999_number_disappear_on_created_order_window\n    self.wait_for_order_number_change(MainPageLocators.ORDER_NUMBER_ON_ORDER_CREATED_WINDOW)\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = <pages.main_page.MainPage object at 0x107337810>\nlocator = ('xpath', \"//h2[@class='Modal_modal__title_shadow__3ikwq Modal_modal__title__2L34m text text_type_digits-large mb-8']\"), time = 20\n\n    def wait_for_order_number_change(self, locator, time=20):\n        def _order_number_changed(driver):\n            element = self.find_element_located_visibility(locator)\n            order_number = element.text\n            return order_number != '9999'\n    \n        try:\n            WebDriverWait(self.browser, time).until(_order_number_changed)\n        except TimeoutException:\n>           raise Exception(\"Номер заказа не изменился!\")\nE           Exception: Номер заказа не изменился!\n\npages/base_page.py:69: Exception"}, "steps": [{"name": "Нажатие на кнопку Войти в аккаунт", "status": "passed", "start": 1723716391978, "stop": 1723716392788}, {"name": "Ввод e-mail адреса", "status": "passed", "start": 1723716392789, "stop": 1723716392838}, {"name": "Ввод пароля", "status": "passed", "start": 1723716392838, "stop": 1723716392849}, {"name": "Нажатие на кнопку Вход", "status": "passed", "start": 1723716392849, "stop": 1723716393097}, {"name": "Добавление ингредиента Флюоресцентная булка R2-D3 в заказ", "status": "passed", "start": 1723716393097, "stop": 1723716394703}, {"name": "Нажатие на кнопку Оформить заказ", "status": "passed", "start": 1723716394703, "stop": 1723716394941}, {"name": "Ожидание пока номера заказа 9999 пропадет и появится актуальный номер заказа в окне созданного заказа", "status": "broken", "statusDetails": {"message": "Exception: Номер заказа не изменился!\n", "trace": "  File \"/Users/ilyazolotov/PycharmProjects/page_object_pythonProject/.venv/lib/python3.11/site-packages/allure_commons/_allure.py\", line 192, in impl\n    return func(*a, **kw)\n           ^^^^^^^^^^^^^^\n  File \"/Users/ilyazolotov/PycharmProjects/Diplom_Ilya_Zolotov_10/Diplom_3/pages/main_page.py\", line 70, in wait_for_9999_number_disappear_on_created_order_window\n    self.wait_for_order_number_change(MainPageLocators.ORDER_NUMBER_ON_ORDER_CREATED_WINDOW)\n  File \"/Users/ilyazolotov/PycharmProjects/Diplom_Ilya_Zolotov_10/Diplom_3/pages/base_page.py\", line 69, in wait_for_order_number_change\n    raise Exception(\"Номер заказа не изменился!\")\n"}, "start": 1723716394942, "stop": 1723716415418}], "parameters": [{"name": "browser", "value": "'firefox'"}], "start": 1723716390176, "stop": 1723716415425, "uuid": "863f8226-4ac9-4d16-a99e-ee03a93e8b78", "historyId": "107e6bc01279cf2957fa148dedc238f0", "testCaseId": "8cd1943bc98f84964817aef1ed87fcfe", "fullName": "tests.test_order_feed_page.TestOrderFeedPage#test_check_order_number_in_the_work_panel_after_making_order", "labels": [{"name": "parentSuite", "value": "tests"}, {"name": "suite", "value": "test_order_feed_page"}, {"name": "subSuite", "value": "TestOrderFeedPage"}, {"name": "host", "value": "MacBook-Pro-Ila.local"}, {"name": "thread", "value": "38882-MainThread"}, {"name": "framework", "value": "pytest"}, {"name": "language", "value": "cpython3"}, {"name": "package", "value": "tests.test_order_feed_page"}]}